lib/non_production_interceptor.rb

```diff
@@ -0,0 +1,11 @@
+# frozen_string_literal: true
+
+class NonProductionInterceptor
+  def self.delivering_email(message)
+    email = Rails.application.secrets.notify_non_production_allow_list
+    body_prefix = "Intended recipient: #{message.to}\n"
```

Comment:
Do we expect message.to to be a string or an array? 

I ask as it looks like we set it to an array in the next statement but seem to be treating it here as a string.

Reply:
You can set `mail.to` to a string or an array, and when you call it will return an array https://www.rubydoc.info/github/mikel/mail/Mail/Message#to-instance_method

Reply:
I'll make a note of this in the commit message as it's not very clear.

Reply:
Sounds good - do we want to do a conversion of the array to a string for the interpolation?

Reply:
I've called join on this so that it can handle multiple email addresses, should we decide to add more to the allow list at a later date.